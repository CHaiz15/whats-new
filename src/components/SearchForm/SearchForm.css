.SearchForm {
  margin: 5px;
  width: 30%;
}

.search-container {
  display: flex;
  justify-content: center;
}

h3 {
  color: white;
}
