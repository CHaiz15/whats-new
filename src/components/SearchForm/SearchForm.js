import React from 'react';
import './SearchForm.css';

const SearchForm = ({searchArticles, searchValue, updateSearchInput}) => {
  return (
    <form className='search-container'>
      <h3>Search Articles:</h3>
      <input
      className='SearchForm'
      type='text'
      placeholder='Type here...'
      value={searchValue}
      onChange={updateSearchInput}
      />
      <button onClick={(event) => searchArticles(searchValue, event)}>Search</button>
    </form>
  )
}

export default SearchForm;
