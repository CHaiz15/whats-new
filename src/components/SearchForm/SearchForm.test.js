import React from 'react'
import { shallow } from 'enzyme'
import SearchForm from './SearchForm.js'

describe('SearchForm', () => {
  it('renders an indication of searchForm', () => {
    const wrapper = shallow(<SearchForm
      searchArticles={jest.fn()}
      searchValue={''}
      updateSearchInput={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  })
  it('should call the searchArticles prop with the search value when the button is clicked', () => {
    const mockSearchArticles = jest.fn();
    const wrapper = shallow(
      <SearchForm
        searchArticles={mockSearchArticles}
        searchValue={'Car'}
        updateSearchInput={jest.fn()}
      />
    );
    wrapper.find('button').simulate('click');
    expect(mockSearchArticles).toHaveBeenCalledWith('Car');
  })
})
