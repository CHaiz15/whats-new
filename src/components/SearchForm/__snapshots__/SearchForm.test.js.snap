// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders an indication of searchForm 1`] = `
<form
  className="search-container"
>
  <h3>
    Search Articles:
  </h3>
  <input
    className="SearchForm"
    onChange={[MockFunction]}
    placeholder="Type here..."
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
